scripts/etc/init.d/rcS

```diff
@@ -528,7 +528,13 @@ echo "OK"
 # default hostname
 echo -n "  Configuring hostname... "
 echo -n $hostname > /rootfs/etc/hostname || fail
-echo "127.0.0.1	localhost" > /rootfs/etc/hosts || fail
+if [ ! -f /rootfs/etc/hosts ]; then
+    echo "127.0.0.1	localhost" > /rootfs/etc/hosts || fail
+else
+    if ! grep -q "localhost" /rootfs/etc/hosts; then
+        echo "127.0.0.1	localhost" >> /rootfs/etc/hosts
+    fi
```

Comment:
IPv6 isn't included here yet, no... :wink: 

also, this is a very simple check to see if a package that was installed, already created the `localhost` entry; so a simple check for the presence of a `/etc/hosts` file and if `localhost` exists in it, should be more than enough...